tests/deploy/e2e/test_deploy.py

```diff
@@ -1068,3 +1068,78 @@ async def test_skipped_for_dependency(resource_container, server, client, client
         deployment_result=DeploymentResult.DEPLOYED,
         blocked=BlockedStatus.NO,
     )
+
+
+async def test_redeploy_after_dependency_recovered(resource_container, server, client, clienthelper, environment, agent):
+    """
+    Asserts that a transiently skipped resource recovers when its dependencies succeed
+    """
+    version = await clienthelper.get_version()
+
+    # Disable auto deploys and repairs
+    Config.set("config", "agent-repair-interval", "0")
+    Config.set("config", "agent-deploy-interval", "0")
+
+    resource_container.Provider.set_fail("agent1", "key", 1)
+
+    rid1 = "test::Resource[agent1,key=key]"
+    rid2 = "test::Resource[agent1,key=key2]"
+
+    resources = [
+        {
+            "key": "key",
+            "value": "value",
+            "id": f"{rid1},v={version}",
+            "requires": [],
+            "purged": False,
+            "send_event": True,
+            "receive_events": False,
+        },
+        {
+            "key": "key2",
+            "value": "value",
+            "id": f"{rid2},v={version}",
+            "requires": [rid1],
+            "purged": False,
+            "send_event": True,
+            "receive_events": True,
+        },
+    ]
+    await clienthelper.set_auto_deploy(True)
+    await clienthelper.put_version_simple(resources, version, wait_for_released=True)
+
+    await clienthelper.wait_for_deployed()
+    scheduler = agent.scheduler
+    assert scheduler._state.resource_state[rid2] == ResourceState(
+        status=ComplianceStatus.NON_COMPLIANT,
+        deployment_result=DeploymentResult.SKIPPED,
+        blocked=BlockedStatus.TRANSIENT,
+    )
+    assert scheduler._state.resource_state[rid1] == ResourceState(
+        status=ComplianceStatus.NON_COMPLIANT,
+        deployment_result=DeploymentResult.FAILED,
+        blocked=BlockedStatus.NO,
+    )
+
+    # Trigger deploy without incrementing version
+    result = await client.deploy(environment)
```

Comment:
This is a tough one. I think that with the current implementation of `deploy` this happens to assert what we want it to. But only as a result of an implementation detail, namely that deploy does not schedule skipped-for-dependencies resources.

What we're really looking to verify is that even if we trigger a deploy *only* for r1, does r2 still get deployed as a result? So I think we need to trigger this more fine-grained deploy. Alas, we don't (yet) expose it on the client. But you already have a scheduler object here, so I'd simply call its `deploy_resource` method.